src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -594,7 +596,7 @@ partial class BoundIncrementOperator : IIncrementExpression
 
     partial class BoundBadExpression
     {
-        protected override OperationKind ExpressionKind => OperationKind.None;
+        protected override OperationKind ExpressionKind => OperationKind.BadExpression;
```

Comment:
OK, how about "InvalidExpression" and "InvalidStatement" operation kinds, and "IsInvalid" for the new property?